### ISSUE TITLE
torch.nn.CrossEntropyLoss slower than custom loss on multichip

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-blacksmith#327]
### Describe the bug

When using torch's inbuilt cross entropy loss on multichip workloads training step takes ~50% more time than when using our custom cross entropy loss from workaround_utils.

This should be looked into, even though we are fine with using custom loss for now
